### PR TITLE
chore: add appropriate padding and update chain selector to styled

### PR DESCRIPTION
**Pull request by JackShort**
* padding needed to be fixed on chain selector: https://linear.app/uniswap/issue/WEB-3080/fix-padding-and-roundedness-on-the-network-dropdown
* updated the chain selector to styled components also

### CODE REVIEW

File: src/components/NavBar/ChainSelector.tsx [hunk 1]
```diff
@@ -13,24 +14,38 @@ import { useOnClickOutside } from 'hooks/useOnClickOutside'
...
-interface ChainSelectorProps {
-  leftAlign?: boolean
-}
+const ChainSelectorWrapper = styled.div`
+  position: relative;
+`
+
+const ChainSelectorButton = styled(BaseButton)<{ isOpen: boolean }>`
+  display: flex;
+  background: ${({ theme, isOpen }) => (isOpen ? theme.accent2 : `none`)};
```
Comment (cbachmeier) [inline src/components/NavBar/ChainSelector.tsx:36]: nit:
```suggestion
  background: ${({ theme, isOpen }) => (isOpen ? theme.accent2 : 'none')};
```

Comment (cbachmeier) [reply src/components/NavBar/ChainSelector.tsx:36]: Also this background color when opened is new, making sure it's intentional?

**Review by cbachmeier — APPROVED** [review]
mostly nits, I think using the ClickableStyle shared style would be a worthwhile improvement

File: src/components/NavBar/ChainSelector.tsx [hunk 2]
```diff
@@ -13,24 +14,38 @@ import { useOnClickOutside } from 'hooks/useOnClickOutside'
...
+  gap: 4px;
+  border-radius: 12px;
+  height: 40px;
+  color: ${({ theme }) => theme.neutral1};
+  transition: ${({ theme }) =>
+    `${theme.transition.duration.medium} ${theme.transition.timing.ease} background-color, ${theme.transition.duration.medium} ${theme.transition.timing.ease} margin`};
+
+  &:hover {
+    background-color: ${({ theme }) => theme.deprecated_stateOverlayHover};
+  }
```
[the remark this answers is not in the document]
Comment (JackShort) [reply src/components/NavBar/ChainSelector.tsx:47]: yeah i tried that initially it didn't look right

File: src/components/NavBar/ChainSelector.tsx (hunk 1 above)
Comment (JackShort) [reply src/components/NavBar/ChainSelector.tsx]: i was also surprised to find that it was in the figma: https://www.figma.com/file/4DJrdsx1lPm5u409xyKDZj/%F0%9F%8D%84-%F0%9F%95%B8%EF%B8%8F-Spore---Web?type=design&node-id=1784%3A36695&mode=design&t=ZjnH9nOnA8LUzT8d-1

also it was part of the old code but wasn't displaying correctly

**Review by mattbspector — APPROVED** [review]

**Pull request merged**